src/ucs/stats/stats.c

```diff
@@ -460,6 +474,171 @@ void ucs_stats_node_free(ucs_stats_node_t *node)
     }
 }
 
+static unsigned
+ucs_stats_agrgt_sum_clsid_alloc(ucs_stats_node_t *node, size_t num_counters_in_class)
+{
+    unsigned offset                      = 0;
+    ucs_stats_filter_node_t *filter_node = node->filter_node;
+    size_t db_size;
+    ucs_stats_node_t *temp_node;
+    uint32_t counter_index;
+    ucs_stats_aggrgt_sum_clsid_to_name_db_t *element;
+    ucs_stats_aggrgt_sum_counter_name_t     *element_cnt_names;
+    ucs_status_t status;
+
+    /* Get current array length */
+    db_size = ucs_array_length(&ucs_stats_context.aggrgt_sum_db);
+
+    status = ucs_array_append(aggrgt_sum_db_array_t, &ucs_stats_context.aggrgt_sum_db);
+    if (status != UCS_OK) {
+        ucs_assert(0);
+        return status;
+    }
+
+    element = ucs_array_last(&ucs_stats_context.aggrgt_sum_db);
+
+    /* Initialize entry */
+    offset = element->aggregate_sum_class_offset = 
+        ucs_array_length(&ucs_stats_context.aggrgt_cnt_names_db);
```

Comment:
```
offset = ucs_array_length(&ucs_stats_context.aggrgt_cnt_names_db);
*ucs_array_last(&ucs_stats_context.aggrgt_sum_db) = offset;
```
Then the variable `element` can be removed

Reply:
Fixed it a little bit differently:

    offset = ucs_array_last(&ucs_stats_context.aggrgt_sum_offsets_db);

    /* Initialize entry */
    *offset = ucs_array_length(&ucs_stats_context.aggrgt_cnt_names_db);